workspaces/global-header/plugins/global-header/src/components/GlobalHeader.tsx

```diff
@@ -0,0 +1,377 @@
+import Button from '@mui/material/Button';
+import { styled } from '@mui/material/styles';
+import InputBase from '@mui/material/InputBase';
+import HelpOutlineOutlinedIcon from '@mui/icons-material/HelpOutlineRounded';
+import NotificationsOutlinedIcon from '@mui/icons-material/NotificationsOutlined';
+import ArrowDropDownOutlinedIcon from '@mui/icons-material/ArrowDropDownOutlined';
+import { Menu } from '@mui/base/Menu';
+import { MenuItem } from '@mui/base/MenuItem';
+import { MenuButton } from '@mui/base/MenuButton';
+import { Dropdown } from '@mui/base/Dropdown';
+import CategoryOutlinedIcon from '@mui/icons-material/CategoryOutlined';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import Typography from '@mui/material/Typography';
+import KeyboardArrowDownOutlinedIcon from '@mui/icons-material/KeyboardArrowDownOutlined';
+import ManageAccountsOutlinedIcon from '@mui/icons-material/ManageAccountsOutlined';
+import LogoutOutlinedIcon from '@mui/icons-material/LogoutOutlined';
+import { Link } from '@backstage/core-components';
+import SearchIcon from '@mui/icons-material/Search';
+import Box from '@mui/material/Box';
+import AccountCircleOutlinedIcon from '@mui/icons-material/AccountCircleOutlined';
+
+const models = [
+  {
+    key: 1,
+    value: 'argocd-template',
+    label: 'Add ArgoCD to an existing project',
+  },
+  {
+    key: 2,
+    value: 'create-backend-plugin',
+    label: 'Create Backend Plugin Template',
+  },
+  {
+    key: 3,
+    value: 'create-frontend-plugin',
+    label: 'Create Frontend Plugin Template',
+  },
+  {
+    key: 4,
+    value: 'create-react-app-template-test-annotator',
+    label: 'Create React App Template with annotations',
+  },
+  {
+    key: 5,
+    value: 'define-ansible-job-template',
+    label: 'Ansible Job Template',
+  },
+];
+
+interface MenuSectionProps {
+  children: React.ReactNode;
+  label: string;
+}
+
+const MenuSectionRoot = styled('li')`
+  list-style: none;
+
+  & > ul {
+    padding-left: 0;
+  }
+`;
+
+const MenuSectionLabel = styled('span')`
+  display: inline-block;
+  font-size: 0.875em;
+  alignment-baseline: baseline;
+  align-content: end;
+  color: #979797;
+`;
+
+function MenuSection({ children, label }: Readonly<MenuSectionProps>) {
+  return (
+    <MenuSectionRoot role="group">
+      <Box
+        style={{
+          display: 'flex',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          margin: '0.75rem',
+        }}
+      >
+        <MenuSectionLabel>{label}</MenuSectionLabel>
+        <MenuItem>
+          <Link
+            underline="none"
+            to="/create"
+            style={{ display: 'inline-block', fontSize: '0.875em' }}
+          >
+            All templates
+          </Link>
+        </MenuItem>
+      </Box>
+      <ul style={{ marginLeft: 12, marginRight: 12 }}>{children}</ul>
+    </MenuSectionRoot>
+  );
+}
+
+export const GlobalHeader = () => {
+  const Search = styled('div')(({ theme }) => ({
```

Comment:
I guess we should have components on the root level. But we should split this file into smaller components. :)